src/coreclr/tools/superpmi/superpmi-shared/asmdumper.cpp

```diff
@@ -19,8 +19,11 @@ void ASMDumper::DumpToFile(HANDLE hFile, MethodContext* mc, CompileResult* cr)
     ZeroMemory(buff, bufflen * sizeof(char));
     buff_offset += sprintf_s(&buff[buff_offset], bufflen - buff_offset,
                              ";;Generated from SuperPMI on original input '%s'", cr->repProcessName());
+
+    char methodName[256];
```

Comment:
I'm a little worried about all these constant `256` around. Can we use something like `printInFull` in these places that automatically get the full string? We need something like a C++ holder helper class to automatically delete memory, or maybe a way to associate the memory with a MethodContext.

Reply:
I introduced `getMethodName` and `getClassName` in spmiutil.cpp that use `std::string` and print the full strings.